### PR TITLE
Updated documentation for vulkan_context.cpp

### PR DESCRIPTION
**Pull request by QuinnLeavitt**
4.x Compliant

This is my first PR please let me know if there are any changes made here that do not comply with documentation standards or if there are any other changes that I should have made to make it more compliant to standards.

I look forward to contributing more in any way that I can. Until I understand the code base better I'll likely stick to fixing comments/documentation.

### CODE REVIEW

**Comment by Calinou** [comment]
> 3.x Compliant

The Vulkan renderer isn't present in `3.x`, so this can't be cherry-picked to `3.x`.

**Comment by QuinnLeavitt** [comment]
Thank you for catching that I changed 3.x to 4.x because of the Vulkan renderer. I forgot about Vulkan not being included in version 3.x.

File: drivers/vulkan/vulkan_context.cpp [hunk 1]
```diff
@@ -2247,13 +2249,13 @@ RID VulkanContext::local_device_create() {
...
 			/*ppEnabledExtensionNames */ (const char *const *)extension_names,
-			/*pEnabledFeatures */ &physical_device_features, // If specific features are required, pass them in here
+			/*pEnabledFeatures */ &physical_device_features, // If specific features are required, pass them in here.
 		};
 		err = vkCreateDevice(gpu, &sdevice, nullptr, &ld.device);
 		ERR_FAIL_COND_V(err, RID());
 	}
 
-	{ //create graphics queue
+	{ /* Create graphics queue. */
```
Comment (aaronfranke) [inline drivers/vulkan/vulkan_context.cpp:2258]: Prefer `//` to `/* */` when writing comments.

Comment (QuinnLeavitt) [reply drivers/vulkan/vulkan_context.cpp:2258]: Is there somewhere that has the exact cases of when to use `//` over `/* */`? It seemed that a lot of places in the code would use `//` for titles of code like the above example but other places would use  `/* */`.

Comment (aaronfranke) [reply drivers/vulkan/vulkan_context.cpp:2258]: https://docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html

Comment (QuinnLeavitt) [reply drivers/vulkan/vulkan_context.cpp:2258]: Thank you that doc is very helpful. I was getting lost in all the different kinds of documentation guidelines last night. Most are for the document pages and not for the code documentation so I couldn't find the rules for comments.

Comment (QuinnLeavitt) [reply drivers/vulkan/vulkan_context.cpp:2258]: Should `//` be used for single line comments inside of if and else statements?

Comment (aaronfranke) [reply drivers/vulkan/vulkan_context.cpp:2258]: When in doubt, use `//`.

Comment (QuinnLeavitt) [reply drivers/vulkan/vulkan_context.cpp:2258]: Is ensuring comments have periods and capitalization necessary? The code style guidelines appear to have periods and capitalization.

Comment (aaronfranke) [reply drivers/vulkan/vulkan_context.cpp:2258]: Yes. There is existing code without periods and capitalization and it's an ongoing effort to improve this.

Comment (QuinnLeavitt) [reply drivers/vulkan/vulkan_context.cpp:2258]: Ok perfect I'll make sure I fix any I come across.

File: drivers/vulkan/vulkan_context.cpp [hunk 2]
```diff
@@ -1192,8 +1192,10 @@ Error VulkanContext::_initialize_queues(VkSurfaceKHR p_surface) {
 		fpGetPhysicalDeviceSurfaceSupportKHR(gpu, i, p_surface, &supportsPresent[i]);
 	}
 
-	// Search for a graphics and a present queue in the array of queue
-	// families, try to find one that supports both
+/**
+* Search for a graphics and a present queue in the array of queue
+* families, try to find one that supports both.
+*/
```
Comment (aaronfranke) [inline drivers/vulkan/vulkan_context.cpp:1198]:
```suggestion
	// Search for a graphics and a present queue in the array of queue
	// families, try to find one that supports both.
```

File: drivers/vulkan/vulkan_context.cpp [hunk 3]
```diff
@@ -1049,19 +1048,20 @@ Error VulkanContext::_create_physical_device(VkSurfaceKHR p_surface) {
...
 	vkGetPhysicalDeviceQueueFamilyProperties(gpu, &queue_family_count, queue_props);
-
-	// Query fine-grained feature support for this device.
-	//  If app has specific feature requirements it should check supported
-	//  features based on this query
+/**
+* Query fine-grained feature support for this device.
+*  If app has specific feature requirements it should check supported
+*  features based on this query.
+*/
```
Comment (aaronfranke) [inline drivers/vulkan/vulkan_context.cpp:1061]:
```suggestion
	// Query fine-grained feature support for this device.
	//  If app has specific feature requirements it should check supported
	//  features based on this query
```

File: drivers/vulkan/vulkan_context.cpp [hunk 4]
```diff
@@ -403,7 +403,7 @@ String VulkanContext::SubgroupCapabilities::supported_stages_desc() const {
 		res += ", STAGE_COMPUTE";
 	}
 
-	/* these are not defined on Android GRMBL */
+	/* These are not defined on Android GRMBL. */
```
Comment (aaronfranke) [inline drivers/vulkan/vulkan_context.cpp:406]:
```suggestion
	// These are not defined on Android GRMBL.
```

**Comment by aaronfranke** [comment]
This PR needs [a squash](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html) before it can be accepted.

**Comment by QuinnLeavitt** [comment]
I don't know if I squashed it correctly.

**Comment by aaronfranke** [comment]
@QuinnLeavitt At the top of the PR it says "Commits `12`". If you squash correctly, it should say "Commits `1`".

**Comment by QuinnLeavitt** [comment]
I can't figure out how to use git rebase to squash my commits, it was telling me there were a bunch of conflicts between files.  What does everyone normally use to squash their commits?

**Comment by aaronfranke** [comment]
@QuinnLeavitt Either read [this article](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html), or follow the instructions in [this video](https://www.youtube.com/watch?v=0XKy8YpP5Ww).

**Comment by QuinnLeavitt** [comment]
Git kraken actually won't let me rebase it because I started rebasing in git bash but I messed the rebase up and it didn't complete correctly and git --abort now doesn't abort the rebase.

**Comment by aaronfranke** [comment]
@QuinnLeavitt If you give me write access to your repo, I can fix the problem for you. Then you can follow the guides if you make future PRs.

**Comment by QuinnLeavitt** [comment]
You should have an invite to collaborate now, I really appreciate all your help and I'll make sure I follow those guides and learn to contribute better.

**Review by aaronfranke — APPROVED** [review]

**Comment by akien-mga** [comment]
Needs a rebase after VRS PR.

**Comment by aaronfranke** [comment]
@QuinnLeavitt You protected your master branch, so I can't push to this PR anymore.

**Comment by aaronfranke** [comment]
Superseded by https://github.com/godotengine/godot/pull/63115

**Pull request closed without merging**